rasa/core/story_conflict.py

```diff
@@ -0,0 +1,252 @@
+        self.correct_response = None
+
+    @staticmethod
+    def find_conflicts(
+        trackers: List[TrackerWithCachedStates], domain: Domain, max_history: int
+    ) -> List:
+        """
+        Generate a list of StoryConflict objects, describing
+        conflicts in the given trackers.
+        :param trackers: Trackers in which to search for conflicts
+        :param domain: The domain
+        :param max_history: The maximum history length to be
+        taken into account
+        :return: List of conflicts
+        """
+
+        # We do this in two steps, to reduce memory consumption:
+
+        # Create a 'state -> list of actions' dict, where the state is
+        # represented by its hash
+        rules = StoryConflict._find_conflicting_states(trackers, domain, max_history)
+
+        # Iterate once more over all states and note the (unhashed) state,
+        # for which a conflict occurs
+        conflicts = StoryConflict._build_conflicts_from_states(
+            trackers, domain, max_history, rules
+        )
+
+        return conflicts
+
+    @staticmethod
+    def _find_conflicting_states(
+        trackers: List[TrackerWithCachedStates], domain: Domain, max_history: int
+    ) -> Dict[int, Optional[List[Text]]]:
+        # Create a 'state -> list of actions' dict, where the state is
+        # represented by its hash
+        rules = {}
+        for tracker, event, sliced_states in StoryConflict._sliced_states_iterator(
+            trackers, domain, max_history
+        ):
+            h = hash(str(list(sliced_states)))
+            if h not in rules:
+                rules[h] = [event.as_story_string()]
+            elif h in rules and event.as_story_string() not in rules[h]:
+                rules[h] += [event.as_story_string()]
+
+        # Keep only conflicting rules
+        return {
+            state_hash: actions
+            for (state_hash, actions) in rules.items()
+            if len(actions) > 1
+        }
+
+    @staticmethod
+    def _build_conflicts_from_states(
+        trackers: List["TrackerWithCachedStates"],
+        domain: Domain,
+        max_history: int,
+        rules: Dict[int, Optional[List[Text]]],
+    ):
+        # Iterate once more over all states and note the (unhashed) state,
+        # for which a conflict occurs
+        conflicts = {}
+        for tracker, event, sliced_states in StoryConflict._sliced_states_iterator(
+            trackers, domain, max_history
+        ):
+            h = hash(str(list(sliced_states)))
+
+            if h in rules and h not in conflicts:
+                conflicts[h] = StoryConflict(sliced_states)
+
+            if h in rules:
+                conflicts[h].add_conflicting_action(
+                    action=event.as_story_string(), story_name=tracker.sender_id
+                )
+
+        # Remove conflicts that arise from unpredictable actions
+        return [c for (h, c) in conflicts.items() if c.has_prior_events]
```

Comment:
I prefer things to be spelled out. To see what `h` means I have to check the code before which is nothing I really wanna do when I am just skimming the code.
```suggestion
        return [c for (hashed_state, conflict) in conflicts.items() if c.has_prior_events]
```